dandiapi/api/models/zarr.py

```diff
@@ -181,6 +181,25 @@ class BaseZarrArchive(TimeStampedModel):
     class Meta:
         get_latest_by = 'modified'
         abstract = True
+        constraints = [
+            models.UniqueConstraint(
+                name='%(app_label)s-%(class)s-unique-name',
```

Comment:
FYI @AlmightyYakob @mvandenburgh 
> You must always specify a unique name for the constraint. As such, you cannot normally specify a constraint on an abstract base class, since the [Meta.constraints](https://docs.djangoproject.com/en/4.0/ref/models/options/#django.db.models.Options.constraints) option is inherited by subclasses, with exactly the same values for the attributes (including name) each time. To work around name collisions, part of the name may contain '%(app_label)s' and '%(class)s', which are replaced, respectively, by the lowercased app label and class name of the concrete model. For example CheckConstraint(check=Q(age__gte=18), name='%(app_label)s_%(class)s_is_adult').

From https://docs.djangoproject.com/en/4.0/ref/models/constraints/.